### PR TITLE
Update virtee/sev dep to 1.2

### PR DESCRIPTION
**Pull request by jepio**
Fix all the API breakage and update az-snp-vtpm dep for compat as well.

### CODE REVIEW

**Review by jialez0 — APPROVED** [review]

**Comment by jepio** [comment]
@jialez0 fixed cargo fmt and rebased due to conflicts in cargo.toml.

File: attestation-service/Cargo.toml [hunk 1]
```diff
@@ -39,7 +39,7 @@ scroll = { version = "0.11.0", default-features = false, features = ["derive"],
 serde.workspace = true
 serde_json.workspace = true
 serde_variant = "0.1.2"
-sev = { version = "= 1.1.0", features = ["openssl"], optional = true }
+sev = { version = "= 1.2.0", features = ["openssl", "snp"], optional = true }
```
Comment (fitzthum) [inline attestation-service/Cargo.toml:42]: Do we still need to set an exact version with these changes?

**Review by fitzthum — APPROVED** [review]
LGTM.

File: attestation-service/Cargo.toml (hunk 1 above)
Comment (jepio) [reply attestation-service/Cargo.toml]: no we don't, i've pushed this change

**Review by sameo — APPROVED** [review]

**Pull request merged**